packages/legend-query-builder/src/components/QueryBuilder.tsx

```diff
@@ -237,131 +243,154 @@ export const QueryBuilder = observer(
         data-testid={QUERY_BUILDER_TEST_ID.QUERY_BUILDER}
         className="query-builder"
       >
+        {queryBuilderState.isEditingWatermark && (
+          <QueryBuilderWatermarkEditor queryBuilderState={queryBuilderState} />
+        )}
+
         <BackdropContainer elementID={QUERY_BUILDER_BACKDROP_CONTAINER_ID} />
         <div className="query-builder__body">
           <div className="query-builder__content">
             <div className="query-builder__sub-header">
-              <div className="query-builder__sub-header__actions">
-                <DropdownMenu
-                  className="query-builder__sub-header__custom-action"
-                  title="Show Advanced Menu..."
-                  content={
-                    <MenuContent>
-                      <MenuContentItem
-                        onClick={toggleShowFunctionPanel}
-                        disabled={!queryBuilderState.isQuerySupported}
-                      >
-                        <MenuContentItemIcon>
-                          {queryBuilderState.showFunctionsExplorerPanel ? (
-                            <CheckIcon />
-                          ) : null}
-                        </MenuContentItemIcon>
-                        <MenuContentItemLabel className="query-builder__sub-header__menu-content">
-                          Show Function(s)
-                        </MenuContentItemLabel>
-                      </MenuContentItem>
-                      {/* TODO?: consider hiding this menu option when the fetch-structure is not TDS */}
-                      {!queryBuilderState.isParameterSupportDisabled && (
+              <div className="query-builder__sub-header__content__icons">
+                {queryBuilderState.watermarkState.value && (
+                  <button
+                    className="panel__header__action"
+                    onClick={openWatermark}
+                    tabIndex={-1}
+                    title="Edit Watermark"
+                  >
+                    <WaterDropIcon />
```

Comment:
Should we have maybe an indicator on the icon if the `forWatermark` function is on for the query. 
Maybe have the icon be a different color if the function is on.

Reply:
I think so to reduce clutters. The `grey` part sounds pretty sad, but that's for now